taipy/core/job/_job_manager.py

```diff
@@ -92,9 +92,14 @@ def _is_deletable(cls, job: Union[Job, JobId]) -> ReasonCollection:
         reason_collector = ReasonCollection()
 
         if isinstance(job, str):
-            job = cls._get(job)
+            job_id = job
+            job = cls._get(job, None)
+        else:
+            job_id = job.id
 
-        if job and not job.is_finished():
+        if job is None:
+            reason_collector._add_reason(job_id, EntityDoesNotExist(job_id))
+        elif not job.is_finished():
```

Comment:
From the previous comment, the case where the job can be None is already handled.
```suggestion
        if not job.is_finished():
```